server/src/test/java/org/elasticsearch/index/mapper/TypeFieldTypeTests.java

```diff
@@ -81,8 +81,31 @@ public void testTermsQuery() throws Exception {
         Mockito.when(mapperService.documentMapper()).thenReturn(mapper);
         query = ft.termQuery("my_type", context);
         assertEquals(new MatchNoDocsQuery(), query);
+        assertWarnings(TypeFieldMapper.TypeFieldType.TYPES_DEPRECATION_MESSAGE);
     }
 
+    public void testExistsQuery() {
+        QueryShardContext context = Mockito.mock(QueryShardContext.class);
+        TypeFieldMapper.TypeFieldType ft = new TypeFieldMapper.TypeFieldType();
+        ft.setName(TypeFieldMapper.NAME);
+        ft.existsQuery(context);
+        assertWarnings(TypeFieldMapper.TypeFieldType.TYPES_DEPRECATION_MESSAGE);
+    }
+
+    public void testRangeQuery() {
+        QueryShardContext context = Mockito.mock(QueryShardContext.class);
+        MapperService mapperService = Mockito.mock(MapperService.class);
+        Mockito.when(mapperService.documentMapper()).thenReturn(null);
```

Comment:
I think we can remove this line, since it's superseded by `Mockito.when(mapperService.documentMapper()).thenReturn(mapper);` below.